dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/service/FileService.java

```diff
@@ -220,6 +229,41 @@ public void exportFiles(OutputStream output, String setId) {
     exportFiles(output, prepResponse, DATA_TABLE_EXPORT_MAP_FIELD_ARRAY);
   }
 
+  public Map<String, Map<String, Long>> getManifestSummary(ManifestSummaryQuery summary) {
+    val response = fileRepository.getManifestSummary(summary);
+
+    val filtersAggs = (InternalFilters) response.getAggregations().asMap().get("summary");
+    val repos = summary.getRepoNames();
+    val responseMap = repos.stream()
+        .map(repo -> {
+          Bucket bucket = filtersAggs.getBucketByKey(repo);
```

Comment:
Maybe this is obvious, but it doesn't seem so at first glance. Mind adding a quick comment to what is happening here?